test/python/pulse/test_builder.py

```diff
@@ -1208,3 +1213,186 @@ def test_call_subroutine_with_parametrized_duration(self):
             pulse.call(subroutine)
 
         self.assertEqual(len(main.blocks), 1)
+
+
+class TestBuilderBaseV2(TestBuilderV2, TestBuilderBase):
+    """Test builder base with backendV2."""
+
+
+class TestContextsV2(TestBuilderV2, TestContexts):
+    """Test builder contexts with backendV2."""
+
+    def test_phase_compensated_frequency_offset(self):
+        """Test that the phase offset context properly compensates for phase
+        accumulation with backendV2."""
+        d0 = pulse.DriveChannel(0)
+        with pulse.build(self.backend) as schedule:
+            with pulse.frequency_offset(1e9, d0, compensate_phase=True):
+                pulse.delay(10, d0)
+
+        reference = pulse.Schedule()
+        reference += instructions.ShiftFrequency(1e9, d0)
+        reference += instructions.Delay(10, d0)
+        reference += instructions.ShiftPhase(
+            -2 * np.pi * ((1e9 * 10 * self.backend.target.dt) % 1), d0
+        )
+        reference += instructions.ShiftFrequency(-1e9, d0)
+        self.assertScheduleEqual(schedule, reference)
+
+
+class TestChannelsV2(TestBuilderV2, TestChannels):
```

Comment:
Is it Ok this implementation ? is it better to split the tests about the builder with backendV2 to another file or not inherit the tests of the builder with backendV1?

Reply:
This is good question. Alternatively we can use data driven test (ddt) to test against both V1 and V2 models (and we need a bit of overhaul of the test). At least you can split the file because the file may become too long if you also mirror other tests for V2.